File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-3.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int l=0, r=0, ans=0, n=s.size(), maxf=0;
        map<char,int>mp;
        while(r<n){
          mp[s[r]]++;
          maxf= max(maxf,mp[s[r]]);
          while(r-l+1-maxf>k){
            mp[s[l]]--;
            l++;
          }
          if(r-l+1-maxf<=k) ans=max(ans, r-l+1);
          r++;
        }
        return ans;
    }
};

File: Data Structures & Algorithms/three-integer-sum/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[][]}
     */
    threeSum(nums) {
        var ans =[];
        let n=nums.length;
        nums.sort((a,b)=>(a-b));
        for(let i=0;i<n;i++){
            if(nums[i]>0) break;
            if(i>0 && nums[i]=== nums[i-1]) continue;
            let j=i+1, k=n-1;
            while(j<k){
                let sum = nums[i]+nums[j]+nums[k];
                if(sum<0) j++;
                else if(sum > 0) k--;
                else {
                    ans.push([nums[i],nums[j],nums[k]]);
                    j++;
                    k--;
                    while(j<k && nums[j] === nums[j-1]) j++;
                    }
            }
        }
        return ans;
    }
}
